state/factory/workingset.go

```diff
@@ -166,6 +166,26 @@ func (ws *workingSet) runAction(
 	if !ok {
 		return nil, errors.New("protocol is empty")
 	}
+	// if it's an EVM tx container, unfold the tx inside
+	if container, ok := selp.Action().(action.EvmTxContainer); ok {
+		var (
+			isContract bool
+			err        error
+		)
+		if container.To() == nil {
+			isContract = true
+		} else {
+			ioAddr, _ := address.FromBytes(container.To().Bytes())
+			isContract, err = ws.checkContract(ctx, ioAddr)
+		}
+		if err != nil {
+			return nil, err
+		}
+		selp, err = container.Unfold(isContract)
```

Comment:
if this is the only place to call `Unfold`, we may simply the interface to
```
  func Unfold(func (*common.Address) bool)) (*SealedEnvelope, error)
```
and delete `func To() *common.Address`.

Reply:
ok, updated